app/scripts/views/settings/recovery_codes.js

```diff
@@ -0,0 +1,151 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+const Cocktail = require('cocktail');
+const FormView = require('../form');
+const ModalSettingsPanelMixin = require('../mixins/modal-settings-panel-mixin');
+const Template = require('templates/settings/recovery_codes.mustache');
+const RecoveryCode = require('../../models/recovery-code');
+
+var t = FormView.t;
+
+const View = FormView.extend({
+  template: Template,
+  className: 'recovery-codes',
+  viewName: 'settings.two-step-authentication.recovery-codes',
+
+  events: {
+    'click .copy-codes': FormView.preventDefaultThen('_copyCodes'),
+    'click .download-codes': FormView.preventDefaultThen('_downloadCodes'),
+    'click .print-codes': FormView.preventDefaultThen('_printCodes'),
+    'click .replace-codes-link': FormView.preventDefaultThen('_replaceRecoveryCodes'),
+    'click .two-step-authentication-done': FormView.preventDefaultThen('_returnToTwoStepAuthentication')
+  },
+
+  _returnToTwoStepAuthentication() {
+    this.navigate('settings/two_step_authentication');
+  },
+
+  _copyCodes() {
+    // This copies the recovery codes to clipboard by creating a tiny transparent
+    // textArea with recovery code contents. Then it executes the
+    // browser `copy` command.
+    // Ref: https://stackoverflow.com/questions/400212/how-do-i-copy-to-the-clipboard-in-javascript
+    const textArea = this.window.document.createElement('textArea');
+    textArea.style.position = 'fixed';
+    textArea.style.top = 0;
+    textArea.style.left = 0;
+    textArea.style.width = '2em';
+    textArea.style.height = '2em';
+    textArea.style.padding = 0;
+    textArea.style.border = 'none';
+    textArea.style.outline = 'none';
+    textArea.style.boxShadow = 'none';
+    textArea.style.background = 'transparent';
+    textArea.value = this.recoveryCodesText;
```

Comment:
Is there a way we could define this bit of markup in a template rather than procedurally in JS?

Reply:
Updated this to use css and jquery instead of document methods, which cleaned it up a lot.